### PR TITLE
Codify reconstruction-aware builder rules

### PR DESCRIPTION
**Pull request by canfieldjuan**
Plan: plans/PR-Coding-For-Reconstruction-Review.md
Slice phase: Workflow/process

This adds the builder-side companion to the already-merged PR reconstruction review protocol. The rule is how to code when that review style is expected: start from the problem and correct-fix shape, keep the diff self-explaining, test behavior, and make the PR body a receipt rather than a shield.

## Intentional
- This does not change the reviewer protocol from #2005; docs/PR_RECONSTRUCTION_PROTOCOL.md remains the reviewer method.
- Docs only. No code, config, workflow, migration, CI gate, detector, or automation is added.

## AI reconciliation
- [chatgpt-codex-connector] Narrow the storage mock allowance: fixed by removing storage from the mock-allowed external-boundary examples and keeping the wording aligned with AGENTS.md real-adapter discipline.
- [chatgpt-codex-connector] Apply the reconstruction check to PR updates too: fixed by changing the AGENTS.md wiring to require the self-check before opening or updating a PR.
- [chatgpt-codex-connector] Do not put shipped behavior changes in Deferred: fixed by separating shipped behavior changes into Scope/Mechanism/PR body and reserving Deferred for split-out follow-up work.

All findings fixed or waived: yes.

## Deferred
- Mechanical enforcement for overclaims, plan/diff drift, or missing self-check evidence remains a separate detector/audit slice.

## Parked hardening
- None.

## Verification
- rg -n "CODING_FOR_RECONSTRUCTION_REVIEW|Code for reconstruction|correct-fix" AGENTS.md CLAUDE.md docs/CODING_FOR_RECONSTRUCTION_REVIEW.md
- python scripts/audit_plan_doc.py plans/PR-Coding-For-Reconstruction-Review.md
- python scripts/audit_plan_doc_files_touched.py plans/PR-Coding-For-Reconstruction-Review.md origin/main
- python scripts/audit_plan_doc_diff_size.py plans/PR-Coding-For-Reconstruction-Review.md origin/main
- python scripts/audit_review_rules_triggered.py origin/main --plan plans/PR-Coding-For-Reconstruction-Review.md
- bash scripts/local_pr_review.sh --current-pr-body-file /tmp/pr-body-coding-for-reconstruction-review.md -- passed after AI reconciliation.

## Diff size
4 files, 189 LOC.

### CODE REVIEW

File: docs/CODING_FOR_RECONSTRUCTION_REVIEW.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+   change, split it out or name it in the plan, PR body, and Deferred section.
+
+4. **Fix the upstream cause.** Do not patch the child symptom just because it is
+   the line the reviewer or CI noticed. If the root is broader than this slice,
+   fix the correct bounded layer and name the remaining upstream work.
+
+5. **Test the behavior, not the story.** Add the happy path and the edge case
+   that would expose the bug or contract drift. Use real local adapters when
+   they exist; mock only external boundaries such as network, time, storage, or
+   third-party APIs.
```
Comment (canfieldjuan) [inline docs/CODING_FOR_RECONSTRUCTION_REVIEW.md]: CONFIRMED (Codex is right): listing "storage" as a mockable boundary contradicts the standing real-adapter rule (§3e.1 requires real service + adapter coverage for persistence-dependent behavior — mock only the true external network edge, use a real/in-memory DB, not mocked storage). This is the exact prefer-real-adapters principle the operator has emphasized; a builder following this line would mock storage and undercut §3e.1. Fix: drop "storage" from the mock-allowed list — mock only network / third-party / clock; storage gets a real adapter.

File: AGENTS.md [hunk 2]
```diff
@@ -256,6 +256,15 @@ If the plan changes mid-implementation (you discovered something the
 plan missed), update the plan doc in the same commit. The plan and
 code ship together.
 
+Code for independent reconstruction
+(`docs/CODING_FOR_RECONSTRUCTION_REVIEW.md`). Before opening a PR, check the
+three-way match yourself: what the diff actually does, what a correct fix for
```
Comment (canfieldjuan) [inline AGENTS.md]: CONFIRMED: the self-check is wired "before opening a PR," but the doc rule (7) says before opening OR updating. Fix loops (re-pushes after review) are exactly where diff-vs-body drift creeps in, so the self-check must re-run on update too. Match the wiring to "opening or updating."

**Review by canfieldjuan — COMMENTED** [review]
Reconstructed from the diff. The CODING_FOR_RECONSTRUCTION_REVIEW.md contract is a sound builder-side complement to the review protocol — 7 rules, PR-body-as-receipt, self-check before push. The 2 open Codex threads are both valid consistency gaps with the existing contract; inline.
